demo/Component_B/component_B.cpp

```diff
@@ -0,0 +1,58 @@
+#include <component.hpp>
+#include <iostream>
+#include <local_communicator.hpp>
+#include <local_component_routing_table.hpp>
+#include <messages/local/local_hello.h>
+#include <messages/spa/subscription_request.h>
+#include <socket/clientSocket.hpp>
+#include "../demo_addresses.hpp"
+
+void messageCallback(cubiumClientSocket_t* sock)
+{
+    SpaMessage* message = (SpaMessage*)sock->buf;
+    std::cout << "Received SpaMessage with opcode: " << (int)message->spaHeader.opcode << '\n';
+    return;
+}
+
+
+class ComponentB : public Component
+{
+public:
+  ComponentB(std::shared_ptr<SpaCommunicator> com = nullptr) : Component(com) {}
+
+  virtual void handleSpaData(SpaMessage*) {}
+  virtual void sendSpaData(LogicalAddress) {}
+
+  virtual void appInit()
+  {
+    std::cout << "Component B initializing!" << '\n';
+
+    LocalHello hello(0, 0, la_LSM, la_CA, 0, 0, 0, 0);
+
+    communicator->getLocalCommunicator()->clientConnect((SpaMessage*)&hello, sizeof(hello), messageCallback);
+
+    SubscriptionRequest request(la_CB, la_CA, la_LSM);
+
+    std::cout << "Sending message with opcode: " << (int)request.spaMessage.spaHeader.opcode << "\n";
```

Comment:
Let's also scrap this in Component B for the moment and just get Component A subscribing to Component B